File: src/main/java/com/hank/Hello.java
package com.hank;


import com.hank.kolin.Student;

public class Hello {
    public static void main(String[] args) {
        Student stu = new Student("ja",66,90);
        stu.highest();
        stu.grading();
//        System.out.println("Hello!");
        String s = "abcde";
//        String s = null;
        System.out.println(s.charAt(3));
        System.out.println(s.length());
        System.out.println(s.substring(3));
        System.out.println(s.substring(1,4));
        Person p = new Person("Tom",66.5f,1.7f);
        p.hello();
        System.out.println("bmi = " + p.bmi());
        int score = 88;
        System.out.println(score < 80 || score > 90);
        char c = 'A';
        System.out.println(c > 'a');

//        dataVariable();

    }

    private static void dataVariable() {
        int age = 19; //32bits
        Integer age2 = 19;
        age2.byteValue();
        System.out.println("中國"+ age2);
        char c = '我'; //16bits
        Character c2 = 'c';
        byte b = 120; //8bits
        float weight = 66.5f; //32bits
        double height = 1.7; //64bits
        boolean adult = true;
        boolean enroll = false;
        String name = "Tom";
    }
}

File: src/main/kotlin/com/hank/Hello.kt
package com.hank

fun main() {
//    println("Hello Kotlin!")
//    var s = "abcde"
    var s:String? = "abcde"
    s = null
//    println(s!!.length)
    println(s?.length)
    println(s?.get(3))
    println(s?.substring(3))
    println(s?.substring(1,4))
    val h = Human(66.5f,1.7f)
    h.hello()
    println(h.bmi())
    val score:Int = 88
    println(score > 60)
    var c : Char = 'A'
    println(c.toInt() > 60)
    /*var age : Int = 19
    age = 20
    var weight = 66.5f // 等於 weight : Float = 66.5f
    var name : String
    name = "hank"*/
}

class Human(var weight:Float, var height:Float, var name:String = "") {
    init {
        println("Init OK. w = $weight , h = $height")
    }
//    constructor(name:String, weight:Float, height:Float):this(weight,height)

    fun bmi():Float {
        val bmi:Float = weight / (height*height)
        return bmi
    }

    fun hello() {
        println("您好！Hello kotlin")
    }
}
